### ISSUE TITLE
[BUG] volumes occasionally enter faulted state

### ISSUE THREAD

**Issue by jemand771** [longhorn/longhorn#3824]
## Describe the bug

Every couple of hours, some (or all) of my longhorn volumes go into a faulted state. While they recover automatically after a few minutes, this still causes workload disruption (auto-delete pod setting) and is generally "inconvenient"

While there are numerous GitHub issues about similar behavior, none of them apply here. Often times the errors were caused by network interruptions, which are unlikely for a single-node cluster.

## To Reproduce
I couldn't reproduce the issue locally and I don't have an identical server to test this on.
If this doesn't go anywhere, I might rent another one for testing purpuoses.

## Expected behavior

Volumes should stay attached and healthy over long times

## Log or Support bundle

screenshot of the event log for one of the affected volumes:
![image](https://user-images.githubusercontent.com/19669567/162061612-6448ce66-fa58-46a5-903b-041f7e561f91.png)

I'm also noticing a lot of similar messages in dmesg:
```
[Mi Apr  6 21:23:24 2022] device veth5e38ea82 left promiscuous mode
[Mi Apr  6 21:23:24 2022] cni0: port 14(veth5e38ea82) entered disabled state
[Mi Apr  6 21:23:24 2022] cni0: port 8(veth0546c1c7) entered disabled state
[Mi Apr  6 21:23:24 2022] device veth0546c1c7 left promiscuous mode
[Mi Apr  6 21:23:24 2022] cni0: port 8(veth0546c1c7) entered disabled state
[Mi Apr  6 21:23:24 2022] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
[Mi Apr  6 21:23:24 2022] cni0: port 8(vethf0e6f9cd) entered blocking state
[Mi Apr  6 21:23:24 2022] cni0: port 8(vethf0e6f9cd) entered disabled state
[Mi Apr  6 21:23:24 2022] device vethf0e6f9cd entered promiscuous mode
[Mi Apr  6 21:23:24 2022] cni0: port 8(vethf0e6f9cd) entered blocking state
[Mi Apr  6 21:23:24 2022] cni0: port 8(vethf0e6f9cd) entered forwarding state
[Mi Apr  6 21:23:24 2022] cni0: port 29(veth840db05b) entered disabled state
[Mi Apr  6 21:23:24 2022] device veth840db05b left promiscuous mode
[Mi Apr  6 21:23:24 2022] cni0: port 29(veth840db05b) entered disabled state
[Mi Apr  6 21:23:24 2022] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
[Mi Apr  6 21:23:24 2022] cni0: port 22(veth9c4acc22) entered disabled state
[Mi Apr  6 21:23:24 2022] device veth9c4acc22 left promiscuous mode
[Mi Apr  6 21:23:24 2022] cni0: port 22(veth9c4acc22) entered disabled state
[Mi Apr  6 21:23:24 2022] cni0: port 14(veth44f23197) entered blocking state
[Mi Apr  6 21:23:24 2022] cni0: port 14(veth44f23197) entered disabled state
[Mi Apr  6 21:23:24 2022] device veth44f23197 entered promiscuous mode
[Mi Apr  6 21:23:24 2022] cni0: port 14(veth44f23197) entered blocking state
[Mi Apr  6 21:23:24 2022] cni0: port 14(veth44f23197) entered forwarding state
[Mi Apr  6 21:23:24 2022] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
[Mi Apr  6 21:23:25 2022] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
[Mi Apr  6 21:23:25 2022] cni0: port 22(vethb2748d9d) entered blocking state
[Mi Apr  6 21:23:25 2022] cni0: port 22(vethb2748d9d) entered disabled state
[Mi Apr  6 21:23:25 2022] device vethb2748d9d entered promiscuous mode
[Mi Apr  6 21:23:25 2022] cni0: port 22(vethb2748d9d) entered blocking state
[Mi Apr  6 21:23:25 2022] cni0: port 22(vethb2748d9d) entered forwarding state
[Mi Apr  6 21:23:25 2022] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
[Mi Apr  6 21:23:25 2022] cni0: port 29(veth55145498) entered blocking state
[Mi Apr  6 21:23:25 2022] cni0: port 29(veth55145498) entered disabled state
[Mi Apr  6 21:23:25 2022] device veth55145498 entered promiscuous mode
[Mi Apr  6 21:23:25 2022] cni0: port 29(veth55145498) entered blocking state
[Mi Apr  6 21:23:25 2022] cni0: port 29(veth55145498) entered forwarding state
[Mi Apr  6 21:23:25 2022] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
[Mi Apr  6 21:23:25 2022] cni0: port 52(veth7110161a) entered blocking state
[Mi Apr  6 21:23:25 2022] cni0: port 52(veth7110161a) entered disabled state
[Mi Apr  6 21:23:25 2022] device veth7110161a entered promiscuous mode
[Mi Apr  6 21:23:25 2022] cni0: port 52(veth7110161a) entered blocking state
[Mi Apr  6 21:23:25 2022] cni0: port 52(veth7110161a) entered forwarding state
[Mi Apr  6 21:23:25 2022] cni0: port 54(vethfa417913) entered disabled state
[Mi Apr  6 21:23:25 2022] device vethfa417913 left promiscuous mode
[Mi Apr  6 21:23:25 2022] cni0: port 54(vethfa417913) entered disabled state
[Mi Apr  6 21:23:34 2022] cni0: port 53(vetha698d128) entered disabled state
[Mi Apr  6 21:23:34 2022] device vetha698d128 left promiscuous mode
[Mi Apr  6 21:23:34 2022] cni0: port 53(vetha698d128) entered disabled state
[Mi Apr  6 21:59:59 2022] IPv6: ADDRCONF(NETDEV_CHANGE): veth6f538fd9: link becomes ready
[Mi Apr  6 21:59:59 2022] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
[Mi Apr  6 21:59:59 2022] cni0: port 53(veth6f538fd9) entered blocking state
[Mi Apr  6 21:59:59 2022] cni0: port 53(veth6f538fd9) entered disabled state
[Mi Apr  6 21:59:59 2022] device veth6f538fd9 entered promiscuous mode
[Mi Apr  6 21:59:59 2022] cni0: port 53(veth6f538fd9) entered blocking state
[Mi Apr  6 21:59:59 2022] cni0: port 53(veth6f538fd9) entered forwarding state
[Mi Apr  6 22:00:00 2022] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
[Mi Apr  6 22:00:00 2022] cni0: port 54(veth50893947) entered blocking state
[Mi Apr  6 22:00:00 2022] cni0: port 54(veth50893947) entered disabled state
[Mi Apr  6 22:00:00 2022] device veth50893947 entered promiscuous mode
[Mi Apr  6 22:00:00 2022] cni0: port 54(veth50893947) entered blocking state
[Mi Apr  6 22:00:00 2022] cni0: port 54(veth50893947) entered forwarding state
[Mi Apr  6 22:00:04 2022] cni0: port 54(veth50893947) entered disabled state
[Mi Apr  6 22:00:04 2022] device veth50893947 left promiscuous mode
[Mi Apr  6 22:00:04 2022] cni0: port 54(veth50893947) entered disabled state
[Mi Apr  6 22:00:06 2022] cni0: port 53(veth6f538fd9) entered disabled state
[Mi Apr  6 22:00:06 2022] device veth6f538fd9 left promiscuous mode
[Mi Apr  6 22:00:06 2022] cni0: port 53(veth6f538fd9) entered disabled state
```

followed by errors that are related to the volume fault
```
[Mi Apr  6 22:00:42 2022] sd 14:0:0:1: [sdm] tag#60 FAILED Result: hostbyte=DID_OK driverbyte=DRIVER_SENSE
[Mi Apr  6 22:00:42 2022] sd 14:0:0:1: [sdm] tag#60 Sense Key : Medium Error [current]
[Mi Apr  6 22:00:42 2022] sd 14:0:0:1: [sdm] tag#60 Add. Sense: Unrecovered read error
[Mi Apr  6 22:00:42 2022] sd 14:0:0:1: [sdm] tag#60 CDB: Write(10) 2a 00 00 10 c1 00 00 00 10 00
[Mi Apr  6 22:00:42 2022] blk_update_request: critical medium error, dev sdm, sector 1097984 op 0x1:(WRITE) flags 0x800 phys_seg 2 prio class 0
[Mi Apr  6 22:00:42 2022] Aborting journal on device sdm-8.
[Mi Apr  6 22:00:42 2022] sd 14:0:0:1: [sdm] tag#61 FAILED Result: hostbyte=DID_OK driverbyte=DRIVER_SENSE
[Mi Apr  6 22:00:42 2022] sd 14:0:0:1: [sdm] tag#61 Sense Key : Medium Error [current]
[Mi Apr  6 22:00:42 2022] sd 14:0:0:1: [sdm] tag#61 Add. Sense: Unrecovered read error
[Mi Apr  6 22:00:42 2022] sd 14:0:0:1: [sdm] tag#61 CDB: Write(10) 2a 00 00 04 50 18 00 00 20 00
[Mi Apr  6 22:00:42 2022] blk_update_request: critical medium error, dev sdm, sector 282648 op 0x1:(WRITE) flags 0x0 phys_seg 4 prio class 0
[Mi Apr  6 22:00:42 2022] EXT4-fs warning (device sdm): ext4_end_bio:309: I/O error 7 writing to inode 16 (offset 0 size 0 starting block 35335)
[Mi Apr  6 22:00:42 2022] Buffer I/O error on device sdm, logical block 35331
[Mi Apr  6 22:00:42 2022] Buffer I/O error on device sdm, logical block 35332
[Mi Apr  6 22:00:42 2022] Buffer I/O error on device sdm, logical block 35333
[Mi Apr  6 22:00:42 2022] Buffer I/O error on device sdm, logical block 35334
[Mi Apr  6 22:00:42 2022] sd 14:0:0:1: [sdm] tag#62 FAILED Result: hostbyte=DID_OK driverbyte=DRIVER_SENSE
[Mi Apr  6 22:00:42 2022] sd 14:0:0:1: [sdm] tag#62 Sense Key : Medium Error [current]
[Mi Apr  6 22:00:42 2022] sd 14:0:0:1: [sdm] tag#62 Add. Sense: Unrecovered read error
[Mi Apr  6 22:00:42 2022] sd 14:0:0:1: [sdm] tag#62 CDB: Write(10) 2a 00 00 04 b0 00 00 00 08 00
[Mi Apr  6 22:00:42 2022] blk_update_request: critical medium error, dev sdm, sector 307200 op 0x1:(WRITE) flags 0x0 phys_seg 1 prio class 0
[Mi Apr  6 22:00:42 2022] EXT4-fs warning (device sdm): ext4_end_bio:309: I/O error 7 writing to inode 16 (offset 0 size 0 starting block 38401)
[Mi Apr  6 22:00:42 2022] Buffer I/O error on device sdm, logical block 38400
```
...and so on, and later:
```
[Mi Apr  6 22:01:09 2022] EXT4-fs warning (device sdm): htree_dirblock_to_tree:1004: inode #2: lblock 0: comm chown: error -5 reading directory block
[Mi Apr  6 22:01:10 2022] sd 7:0:0:1: [sdf] Synchronizing SCSI cache
[Mi Apr  6 22:01:10 2022] JBD2: Error while async write back metadata bh 1.
[Mi Apr  6 22:01:10 2022] JBD2: Error while async write back metadata bh 1033.
[Mi Apr  6 22:01:10 2022] JBD2: Error while async write back metadata bh 64513.
[Mi Apr  6 22:01:10 2022] sd 5:0:0:1: [sdd] Synchronizing SCSI cache
[Mi Apr  6 22:01:10 2022] sd 13:0:0:1: [sdl] Synchronizing SCSI cache
[Mi Apr  6 22:01:12 2022] EXT4-fs warning (device sdl): dx_probe:768: inode #393224: lblock 0: comm rspamd: error -5 reading directory block
[Mi Apr  6 22:01:12 2022] EXT4-fs error (device sdl): __ext4_find_entry:1538: inode #393223: comm rspamd: reading directory lblock 0
[Mi Apr  6 22:01:12 2022] EXT4-fs warning (device sdl): dx_probe:768: inode #393224: lblock 0: comm rspamd: error -5 reading directory block
[Mi Apr  6 22:01:12 2022] EXT4-fs error (device sdl): __ext4_find_entry:1538: inode #393223: comm rspamd: reading directory lblock 0
[Mi Apr  6 22:01:12 2022] EXT4-fs warning (device sdl): dx_probe:768: inode #393224: lblock 0: comm rspamd: error -5 reading directory block
[Mi Apr  6 22:01:12 2022] EXT4-fs warning (device sdl): dx_probe:768: inode #393224: lblock 0: comm rspamd: error -5 reading directory block
[Mi Apr  6 22:01:12 2022] EXT4-fs error (device sdl): __ext4_find_entry:1538: inode #393223: comm rspamd: reading directory lblock 0
[Mi Apr  6 22:01:12 2022] EXT4-fs error (device sdl): __ext4_find_entry:1538: inode #393223: comm rspamd: reading directory lblock 0
[Mi Apr  6 22:01:12 2022] EXT4-fs warning (device sdl): dx_probe:768: inode #393224: lblock 0: comm rspamd: error -5 reading directory block
[Mi Apr  6 22:01:12 2022] EXT4-fs error (device sdl): __ext4_find_entry:1538: inode #393223: comm rspamd: reading directory lblock 0
[Mi Apr  6 22:01:12 2022] EXT4-fs error (device sdl): __ext4_find_entry:1538: inode #393223: comm rspamd: reading directory lblock 0
[Mi Apr  6 22:01:12 2022] EXT4-fs warning (device sdl): dx_probe:768: inode #393224: lblock 0: comm rspamd: error -5 reading directory block
```
...and so on

support bundle:
[longhorn-support-bundle_f836c76b-edfa-4967-bc59-e6be3cd15ba1_2022-04-06T20-18-24Z.zip](https://github.com/longhorn/longhorn/files/8430301/longhorn-support-bundle_f836c76b-edfa-4967-bc59-e6be3cd15ba1_2022-04-06T20-18-24Z.zip)

note that my timezone is CEST/GMT+2 (Europe/Berlin with dst) and I'm not sure what timezone the bundle uses. My dmesg excerpt uses gmt+2 (around 10pm)m so 8pm utc.

## Environment

 - Longhorn version: 1.2.4
 - Installation method kubectl
 - Kubernetes distro and version: k3s v1.22.6+k3s1
   - Number of management node in the cluster: 1
   - Number of worker node in the cluster: 0 (or technically 1, since the manager node is _also_ a worker)
 - Node config
   - OS type and version: Ubuntu 20.04.4 LTS
   - CPU per node: 10
   - Memory per node: 64gb
   - Disk type: SSD, not sure if sata/nvme/...
   - Network bandwidth between the nodes: n/a
 - Underlying Infrastructure: linux "vm" on kvm, (supposedly) dedicated cpu/memory. hosted on netcup.de
 - Number of Longhorn volumes in the cluster: 10

## Additional context

I checked for the multipath issue and added the recommended rule.
It's possible that the issue is somehow related to my setup and/or hosting provider since noone else seems to have experienced this yet. I found a single thread in my hosting provider's forum where a user complained about longhorn causing issues, but that didn't really lead anywhere. (thread [here](https://forum.netcup.de/netcup-intern/operations/13307-rancher-und-longhorn-auf-netcup-servern-aufbauen-erfahrungen/) (german))
I also tried switching through the available "storage drivers" that my provider offers, but the issue occurs on all three of them (isci, virtio, ide)
From grepping through my journal, it looks like I've been getting this ever since I installed longhorn in 1.2.3 (about a month ago). I'm only noticing this now as I slowly migrate my volumes to longhorn.

a few responses to the stuff that has already been identified as the issue for other users:
* inter-node bandwidth: not applicable, it's a single-node cluster.
* intra-node (flannel) connection interruptions: I _think_ I would have noticed something like that. the server in question also runs a small game server on k8s, for which players would probably have complained about interruptions
* disk speed: while I don't know the underlying hardware, I know that it's solid state. should be fast enough
* cpu utilization: while I've set the guaranteed cpu down to 3% (default is 12 iirc), that server has plenty of resources to spare (around 65% cpu usage in peak times). the cluster is heavily overprovisioned, but nothing actually uses as much as its limits suggest

**Comment by shuo-wu** [comment]
After a quick glance, I did see a lot of error messages like followings:
```
2022-04-06T10:25:34.996927843+02:00 [pvc-ea1f1c86-17da-4687-9411-2a52a16edb0e-e-6dd3ded1] time="2022-04-06T08:25:34Z" level=error msg="R/W Timeout. No response received in 8s"

2022-04-06T10:25:35.563817705+02:00 Apr  6 08:18:01.990863: ->10.42.0.54:10001 P[  0[pvc-5bab552c-e225-4218-a539-6975fed4c85b-e-b4142590] time="2022-04-06T08:25:35Z" level=error msg="R/W Timeout. No response received in 8s"

2022-04-06T10:25:35.563817705+02:00 Apr  6 08:18:01.990863: ->10.42.0.54:10001 P[  0[pvc-5bab552c-e225-4218-a539-6975fed4c85b-e-b4142590] time="2022-04-06T08:25:35Z" level=error msg="R/W Timeout. No response received in 8s"
```
```
2022-04-06T10:26:16.289747067+02:00 [pvc-689f23a8-b3a6-4687-98ab-fc5145a669be-r-fbabd664] time="2022-04-06T08:25:49Z" level=error msg="Failed to write: write tcp 10.42.0.54:10031->10.42.0.53:51802: write: broken pipe"
2022-04-06T10:26:16.289750684+02:00 [pvc-ea1f1c86-17da-4687-9411-2a52a16edb0e-r-3e23b6b4] time="2022-04-06T08:25:36Z" level=warning msg="Received signal interrupt to shutdown"
2022-04-06T10:26:16.289753209+02:00 time="2022-04-06T08:25:36Z" level=warning msg="Starting to execute registered shutdown func github.com/longhorn/longhorn-engine/app/cmd.startReplica.func4"
2022-04-06T10:26:16.309222419+02:00 [pvc-a18aad34-f40f-4d18-9b8b-f7e3849a07c0-r-22ac8d48] time="2022-04-06T08:25:49Z" level=error msg="Failed to write: write tcp 10.42.0.54:10016->10.42.0.53:54836: write: broken pipe"
2022-04-06T10:26:16.309311188+02:00 [pvc-5bab552c-e225-4218-a539-6975fed4c85b-r-4cd164bc] time="2022-04-06T08:25:49Z" level=error msg="Failed to write: write tcp 10.42.0.54:10106->10.42.0.53:50832: write: broken pipe"
2022-04-06T10:26:16.309353529+02:00 time="2022-04-06T08:25:49Z" level=error msg="Failed to write: write tcp 10.42.0.54:10106->10.42.0.53:50832: write: broken pipe"
2022-04-06T10:26:16.311125027+02:00 [pvc-9db54526-847a-4327-8e64-a081d945d970-r-5d7a03ec] time="2022-04-06T08:25:49Z" level=error msg="Failed to write: write tcp 10.42.0.54:10046->10.42.0.53:48290: write: broken pipe"
2022-04-06T10:26:16.311255346+02:00 time="2022-04-06T08:26:16Z" level=error msg="Failed to write: write tcp 10.42.0.54:10046->10.42.0.53:48290: write: broken pipe"
2022-04-06T10:26:16.321066647+02:00 [pvc-6a7a43f3-191c-496b-a88a-170fac1ad269-r-f4514538] time="2022-04-06T08:26:07Z" level=error msg="Failed to write: write tcp 10.42.0.54:10076->10.42.0.53:36460: write: broken pipe"
2022-04-06T10:26:16.328183511+02:00 time="2022-04-06T08:26:16Z" level=error msg="Failed to write: write tcp 10.42.0.54:10076->10.42.0.53:36460: write: broken pipe"
2022-04-06T10:26:16.328190013+02:00 time="2022-04-06T08:26:16Z" level=error msg="Failed to write: write tcp 10.42.0.54:10076->10.42.0.53:36460: write: broken pipe"
```

Not sure what happens... Your cluster is a single-node cluster. Is there anything with the cluster network config?

**Comment by jemand771** [comment]
as far as I know, there's nothing unusual/non-default about my network config.
I don't really know where to go from here debugging-wise: I don't see any network-related messages in my journal and everything else seems to be fine. I can understand if you don't want to troubleshoot my cluster for a possibly unrelated problem, but longhorn is the thing that isn't working after all

Note that the other netcup customer resolved their issue by switching from flannel to weave. I'm not sure if it's a coincidence/I'm imagining things or if this might actually be related to my provider. I can choose a driver for my network interface: virtio, e1000 or rtl8139. I don't think this matters (as the network issues seem to be "higher up" inside the cluster), but I'm gonna try switching between those aswell.

tl;dr: no clue

**Comment by shuo-wu** [comment]
YES. It's more likely to be related to the cluster level network issue. Besides, I just rechecked the support bundle and found some new clues:
1. I found that you node/kubelet was down (and recovered immediately) at `2022-04-06T19:18:22Z`. Notice that Longhorn uses UTC time.
  ```
  objectmeta:
    name: nc1.16e364c86445ab43
    generatename: ""
    namespace: longhorn-system
    selflink: ""
    uid: 83af08d3-9019-41c7-a539-b61f44c32770
    resourceversion: "25364970"
    generation: 0
    creationtimestamp: "2022-04-06T19:18:20Z"
    deletiontimestamp: null
    deletiongraceperiodseconds: null
    labels: {}
    annotations: {}
    ownerreferences: []
    finalizers: []
    clustername: ""
  involvedobject:
    kind: Node
    namespace: longhorn-system
    name: nc1
    uid: fe15873f-4a24-43f1-bce0-0f484dd54d34
    apiversion: longhorn.io/v1beta1
    resourceversion: "25361582"
    fieldpath: ""
  reason: Ready
  message: 'Kubernetes node nc1 not ready: KubeletNotReady'
  source:
    component: longhorn-node-controller
    host: ""
  firsttimestamp: "2022-04-06T19:18:20Z"
  lasttimestamp: "2022-04-06T19:18:20Z"
  count: 2
  type: Warning
  ```
2. All replicas were suddenly down at `2022-04-06T20:01:05Z`. This leads to the volume Faulted. As I mentioned before, there were lots of TCP connection broken around `2022-04-06T20:01Z` in the replica manager, which was the direct reason of the replica down.
  ```
  2022-04-06T22:00:42.831088601+02:00 [pvc-6a7a43f3-191c-496b-a88a-170fac1ad269-r-f4514538] time="2022-04-06T20:00:42Z" level=info msg="Lost connection from: 10.42.0.53:51908"
  2022-04-06T22:00:43.431193749+02:00 [pvc-5bab552c-e225-4218-a539-6975fed4c85b-r-4cd164bc] time="2022-04-06T20:00:43Z" level=info msg="Lost connection from: 10.42.0.53:57078"
  2022-04-06T22:00:44.037346377+02:00 [pvc-a18aad34-f40f-4d18-9b8b-f7e3849a07c0-r-22ac8d48] time="2022-04-06T20:00:44Z" level=info msg="Lost connection from: 10.42.0.53:39214"
  2022-04-06T22:00:51.911209375+02:00 [pvc-689f23a8-b3a6-4687-98ab-fc5145a669be-r-fbabd664] time="2022-04-06T20:00:51Z" level=info msg="Lost connection from: 10.42.0.53:48200"
  2022-04-06T22:00:52.924847697+02:00 [pvc-ea1f1c86-17da-4687-9411-2a52a16edb0e-r-3e23b6b4] time="2022-04-06T20:00:52Z" level=info msg="Lost connection from: 10.42.0.53:44352"
  2022-04-06T22:01:05.274753060+02:00 [pvc-ea1f1c86-17da-4687-9411-2a52a16edb0e-r-3e23b6b4] time="2022-04-06T20:00:56Z" level=error msg="Failed to write: write tcp 10.42.0.54:10016->10.42.0.53:44352: write: broken pipe"
  2022-04-06T22:01:05.301764714+02:00 [pvc-6a7a43f3-191c-496b-a88a-170fac1ad269-r-f4514538] time="2022-04-06T20:00:56Z" level=error msg="Failed to write: write tcp 10.42.0.54:10076->10.42.0.53:51908: write: broken pipe"
  2022-04-06T22:01:05.301787608+02:00 time="2022-04-06T20:00:56Z" level=error msg="Failed to write: write tcp 10.42.0.54:10076->10.42.0.53:51908: write: broken pipe"
  2022-04-06T22:01:05.301792767+02:00 time="2022-04-06T20:00:56Z" level=error msg="Failed to write: write tcp 10.42.0.54:10076->10.42.0.53:51908: write: broken pipe"
  2022-04-06T22:01:05.301805652+02:00 [pvc-5bab552c-e225-4218-a539-6975fed4c85b-r-4cd164bc] time="2022-04-06T20:00:56Z" level=error msg="Failed to write: write tcp 10.42.0.54:10001->10.42.0.53:57078: write: broken pipe"
  2022-04-06T22:01:05.301810040+02:00 [pvc-689f23a8-b3a6-4687-98ab-fc5145a669be-r-fbabd664] time="2022-04-06T20:00:56Z" level=error msg="Failed to write: write tcp 10.42.0.54:10106->10.42.0.53:48200: write: broken pipe"
  2022-04-06T22:01:05.301813347+02:00 time="2022-04-06T20:01:05Z" level=error msg="Failed to write: write tcp 10.42.0.54:10106->10.42.0.53:48200: write: broken pipe"
  2022-04-06T22:01:05.301829759+02:00 time="2022-04-06T20:01:05Z" level=error msg="Failed to write: write tcp 10.42.0.54:10106->10.42.0.53:48200: write: broken pipe"
  2022-04-06T22:01:05.301833456+02:00 time="2022-04-06T20:01:05Z" level=error msg="Failed to write: write tcp 10.42.0.54:10106->10.42.0.53:48200: write: broken pipe"
  2022-04-06T22:01:05.301836802+02:00 time="2022-04-06T20:01:05Z" level=error msg="Failed to write: write tcp 10.42.0.54:10106->10.42.0.53:48200: write: broken pipe"
  ```
3. There is no log for the several hours prior to `2022-04-06T20:00:42Z` in engine/replica managers. There shouldn't be nothing happen before this timestamp. And the longhorn manager log somehow is truncated and starts at `2022-04-06T20:01:18Z`, which is abnormal. 
4. In the 1st dmesg log block you posted, I saw lots of the following logs. Do these mean that the cluster network is flapping? Sorry I am not familiar with this part and cannot help a lot.
  ```
  [Mi Apr  6 22:00:00 2022] device veth50893947 entered promiscuous mode
  [Mi Apr  6 22:00:00 2022] cni0: port 54(veth50893947) entered blocking state
  [Mi Apr  6 22:00:00 2022] cni0: port 54(veth50893947) entered forwarding state
  [Mi Apr  6 22:00:04 2022] cni0: port 54(veth50893947) entered disabled state
  [Mi Apr  6 22:00:04 2022] device veth50893947 left promiscuous mode
  [Mi Apr  6 22:00:04 2022] cni0: port 54(veth50893947) entered disabled state
  ```

**Comment by jemand771** [comment]
1: the kubelet being down was caused by my restarting the node, so that should be unrelated
3: I checked grafana loki (collects all container logs) for the engine and replica managers, and neither of them seem to have logged anything before the time you mentioned (-> the info in the support bundle is correct)

on 2+4: someting really seems to be broken about my flannel (or cluster networking in general) but I have no idea what, how, or why D:

**Comment by shuo-wu** [comment]
Maybe you can find and check flannel logs. And I am still confused about why part of the longhorn manager pod log is missed...

**Comment by jemand771** [comment]
another "running out of things to troubleshoot"-update:
* I tried the different network adapter virtualizations I mentioned, with no luck
* k3s doesn't seem to have very verbose flannel logging, so either it doesn't log much or it's all fine
* I came across https://kubernetes.io/blog/2019/03/29/kube-proxy-subtleties-debugging-an-intermittent-connection-reset/ which is a really old issue, but worth a try
  * I enabled `/proc/sys/net/netfilter/nf_conntrack_tcp_be_liberal`, but the issue still occured. It felt like it took a little longer than usual, but that might just be a coincidence
  * I deployed the reproducer from the article, it didn't get triggered. so either it's not that issue (as it has officially been fixed in k8s 1.15), or the network interruption doesn't happen for all pods simultaneously
* I switched the cluster to calico because someone recommendet it (and I was getting sick of flannels nonexistent logs)
  * just swapping out the cluster networking like that isn't officially documented, but it just worked right away (disable flannel, install calico, reboot)
  * the disconnect/timeout still happens
  * none of the calico logs show anything unusual, other than a small "new workload" spam after longhorn has deleted the pods  and the new ones get assigned ip addresses (intended+expected)

so for me, that leaves three possible scenarios:
1. it's something about my host messing with the networking (very likely+hard to debug)
2. The flannel -> calico switch didn't quite work and it's flannel's fault (I don't think so, but I should keep it in mind)
3. it's a longhorn issue after all (also unlikely as the logs indicate a network failure)

I'm probably going to try tcpdump next and see if I can spot anything unusual. However, I don't expect much from just looking at the raw packets :c

**Comment by jemand771** [comment]
I've bit the bullet and reinstalled the machine+cluster, everything works as intended now.
Ideally, I could have just reinstalled the cluster and leave the os as-is to pinpoint the issue, but I really didn't feel like redeploying everything twice :b

Thanks for the help (+the amazing software in general), I'm now 99.9% sure that none of this was longhorn's fault.
